components/ai_chat/core/browser/leo_local_models_updater.h

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_LEO_LOCAL_MODELS_UPDATER_H_
+#define BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_LEO_LOCAL_MODELS_UPDATER_H_
+
+#include <string>
+
+#include "base/files/file_path.h"
+#include "base/observer_list.h"
+#include "base/observer_list_types.h"
+#include "brave/components/brave_component_updater/browser/brave_component.h"
+
+class LeoLocalModelsUpdaterTest;
+
+namespace ai_chat {
+
+extern const char kUniversalQAModelName[];
+
+// A component updater for downloading local models for Leo.
+class LeoLocalModelsUpdater : public brave_component_updater::BraveComponent {
```

Comment:
`BraveComponent` really needs to be deprecated, nothing new should be using it. Please follow standard component installer patterns from upstream using `component_updater::ComponentInstallerPolicy`

Reply:
fixed in 7e443e164c9e4952e57e1d040e9f300d7fa521e2